### PR TITLE
[Storage][Sharding] Support checkpoint on sharded DBs.

### PR DESCRIPTION
**Pull request by grao1991**

### CODE REVIEW

File: storage/aptosdb/src/state_kv_db.rs [hunk 1]
```diff
@@ -137,6 +141,30 @@ impl StateKvDb {
...
+                    .db_shard(shard_id as u8)
+                    .create_checkpoint(Self::db_shard_path(
+                        cp_root_path.as_ref(),
+                        shard_id as u8,
+                    ))?;
+            }
+        }
+        state_kv_db
+            .metadata_db()
+            .create_checkpoint(Self::metadata_db_path(cp_root_path))
```
Comment (msmouse) [inline storage/aptosdb/src/state_kv_db.rs:165]: My intuition: you need to checkpoint the metadata first? (Assuming one day we'll be able to checkpoint on the side while the node is running)

File: storage/aptosdb/src/state_merkle_db.rs [hunk 2]
```diff
@@ -124,10 +140,45 @@ impl StateMerkleDb {
...
+                state_merkle_db
+                    .db_shard(shard_id as u8)
+                    .create_checkpoint(Self::db_shard_path(
+                        cp_root_path.as_ref(),
+                        shard_id as u8,
+                    ))?;
+            }
+        }
+        state_merkle_db
+            .metadata_db()
```
Comment (msmouse) [inline storage/aptosdb/src/state_merkle_db.rs:170]: ditto

File: storage/aptosdb/src/state_kv_db.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/state_kv_db.rs]: Done

File: storage/aptosdb/src/state_merkle_db.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/state_merkle_db.rs]: Done

**Review by msmouse — APPROVED** [review]

**Review by areshand — APPROVED** [review]

**Pull request merged**